clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -4125,6 +4125,37 @@ def MemChrOp : CIR_Op<"libc.memchr"> {
   let hasVerifier = 0;
 }
 
+//===----------------------------------------------------------------------===//
+// ReturnAddrOp
+//===----------------------------------------------------------------------===//
+
+def ReturnAddrOp : CIR_Op<"return_address"> {
+  let arguments = (ins UInt32:$level);
+  let summary = "call to `__builtin_return_address`";
```

Comment:
Summary should be a bit more generic, extracting from your full desc, one idea: `Returns the return address of the current function or of one of its callers`.

Reply:
Totally make sense. Just an annotate: your suggested sentence is a word-2-word match of [GNU's summary for the builtin](https://gcc.gnu.org/onlinedocs/gcc/Return-Address.html)